src/rpc/misc.cpp

```diff
@@ -40,6 +40,52 @@ class DescribeAddressVisitor : public boost::static_visitor<UniValue>
 
     explicit DescribeAddressVisitor(CWallet *_pwallet) : pwallet(_pwallet) {}
 
+    void ProcessSubScript(const CScript& subscript, UniValue& obj, bool include_addresses = false) const
+    {
+        // Always present: script type and redeemscript
+        txnouttype which_type;
+        std::vector<std::vector<unsigned char>> solutions_data;
+        Solver(subscript, which_type, solutions_data);
+        obj.pushKV("script", GetTxnOutputType(which_type));
+        obj.pushKV("hex", HexStr(subscript.begin(), subscript.end()));
+
+        // Only when the script corresponds to an address:
+        CTxDestination embedded;
+        UniValue a(UniValue::VARR);
+        if (ExtractDestination(subscript, embedded)) {
+            UniValue subobj = boost::apply_visitor(*this, embedded);
+            subobj.pushKV("address", EncodeDestination(embedded));
+            subobj.pushKV("scriptPubKey", HexStr(subscript.begin(), subscript.end()));
+            if (subobj.exists("pubkey")) {
+                // Always report the pubkey at the top level, so that `getnewaddress()['pubkey']` always works.
+                obj.pushKV("pubkey", subobj["pubkey"]);
+            }
+            obj.pushKV("embedded", std::move(subobj));
+            a.push_back(EncodeDestination(embedded));
```

Comment:
Could add `if (include_addresses) a.push_back(...);`

Reply:
Why? `embedded` should always be reported, it's not obsolete like `addresses`.

Reply:
Fixed, see elsewhere.